eventauth.go

```diff
@@ -204,25 +191,25 @@ func Allowed(event Event, authEvents AuthEvents) error {
 // createEventAllowed checks whether the m.room.create event is allowed.
 // It returns an error if the event is not allowed.
 func createEventAllowed(event Event) error {
-	if event.StateKey == nil {
+	if !event.IsState() {
 		return errorf("create event missing state key")
 	}
-	if *event.StateKey != "" {
-		return errorf("create event state key is not empty: %q", event.StateKey)
+	if event.StateKey() != "" {
```

Comment:
Or alternatively expose a `StateKeyEquals(str)` function which will handle the non-existent case for you, since most of the time you are doing comparisons.